utils/frame/benchmarking-cli/src/shared/mod.rs

```diff
@@ -73,3 +73,19 @@ pub fn new_rng(seed: Option<u64>) -> (impl rand::Rng, u64) {
 	let seed = seed.unwrap_or(rand::thread_rng().gen::<u64>());
 	(rand_pcg::Pcg64::seed_from_u64(seed), seed)
 }
+
+/// Returns an error if a debug profile is detected.
+///
+/// The rust compiler only exposes the binary information whether
+/// or not we are in a `debug` build.
+/// This means that `release` and `production` cannot be told apart.
+/// This function additionally checks for OPT-LEVEL = 3.
+pub fn check_build_profile() -> Result<(), String> {
+	if cfg!(debug_assertions) {
```

Comment:
`debug_assertions` doesn't mean this is a debug build.

Please use this https://github.com/paritytech/substrate/blob/4a5271dcb2ca09bd8a9d9b3d83502dba03b3c3f3/client/consensus/slots/build.rs#L22-L24 + https://github.com/paritytech/substrate/blob/4a5271dcb2ca09bd8a9d9b3d83502dba03b3c3f3/client/consensus/slots/src/lib.rs#L317